lib/compilers/snowball.ts

```diff
@@ -0,0 +1,108 @@
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+import path from 'path';
+
+import _ from 'underscore';
+
+import type {PreliminaryCompilerInfo} from '../../types/compiler.interfaces.js';
+import type {ParseFiltersAndOutputOptions} from '../../types/features/filters.interfaces.js';
+import {unwrap} from '../assert.js';
+import {BaseCompiler} from '../base-compiler.js';
+
+export class SnowballCompiler extends BaseCompiler {
+    linker: string;
+
+    static get key() {
+        return 'snowball';
+    }
+
+    constructor(info: PreliminaryCompilerInfo, env) {
+        super(info, env);
+        this.compiler.supportsIntel = true;
+        this.compiler.supportsIrView = true;
+        this.compiler.supportsLLVMOptPipelineView = true;
+
+        this.compiler.irArg = ['--emit', 'llvm-ir'];
+        this.linker = this.compilerProps<string>('linker');
+    }
+
+    override getSharedLibraryPathsAsArguments(libraries, libDownloadPath) {
+        return [];
+    }
+
+    override getSharedLibraryLinks(libraries: any[]): string[] {
+        return [];
+    }
+
+    override orderArguments(
+        options: string[],
+        inputFilename: string,
+        libIncludes: string[],
+        libOptions: string[],
+        libPaths: string[],
+        libLinks: string[],
+        userOptions: string[],
+        staticLibLinks: string[],
+    ) {
+        return options.concat(userOptions, libIncludes, libOptions, libPaths, libLinks, staticLibLinks, [
+            this.filename(inputFilename),
+        ]);
+    }
+
+    override optionsForFilter(filters: ParseFiltersAndOutputOptions, outputFilename: string, userOptions?: string[]) {
+        let options = ['build', '-o', this.filename(outputFilename)];
+
+        const userRequestedEmit = _.any(unwrap(userOptions), opt => opt.includes('--emit'));
+        if (filters.binary) {
+            options = options.concat(['--emit', 'exec']);
+        } else if (filters.binaryObject) {
+            options = options.concat(['--emit', 'lib']);
+        } else {
+            if (!userRequestedEmit) {
+                options = options.concat('--emit', 'llvm-ir');
+            }
+            if (!userRequestedEmit) {
+                options = options.concat('--emit', 'asm');
+            }
+            // TODO:
+            // if (filters.intel) options = options.concat('--llvm-args', '--x86-asm-syntax=intel');
```

Comment:
When you say "TODO" - these are not supported? Then I think we need to disable them? YOu can set `compiler.supportsIntel = false` here, or in the config file.

Reply:
I was planning to add support for that in the future. Can it be changed later?

Reply:
It can yes :)